src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -288,7 +292,7 @@ async def run_setup_pys(
         tgt = target_with_origin.target
         if tgt.has_field(PythonProvidesField):
             exported_targets.append(ExportedTarget(tgt))
-        elif isinstance(target_with_origin.origin, AddressLiteralSpec):
+        elif isinstance(target_with_origin.origin, (AddressLiteralSpec, FilesystemLiteralSpec)):
```

Comment:
We should have had this from the start. If any literal spec is used, we will error.